pkg/provider/pipeline.go

```diff
@@ -415,3 +426,13 @@ func resourcePipelineDelete(d *schema.ResourceData, m interface{}) error {
 	d.SetId("")
 	return nil
 }
+
+// return the pieces of id `left:right` as left, right
+func parseTwoPartID(id, left, right string) (string, string, error) {
+	parts := strings.SplitN(id, ":", 2)
+	if len(parts) != 2 {
```

Comment:
Probably worth checking for either of the parts being empty strings too.

FWIW I'm not sure it's worth trying to make this function generic enough to be able to take custom field names for the error messages, it just ends up being confusing what `left` and `right` are actually for.

Reply:
I just copied this from the github provider.